source/sessions/tests/unified/snapshot-sessions.json

```diff
@@ -0,0 +1,698 @@
+          "object": "collection0",
+          "arguments": {
+            "filter": {
+              "_id": 4
+            },
+            "update": {
+              "$inc": {
+                "x": 1
+              }
+            },
+            "returnDocument": "After"
+          },
+          "expectResult": {
+            "_id": 4,
+            "x": 12
+          }
+        },
+        {
+          "name": "find",
+          "object": "collection0",
+          "arguments": {
+            "filter": {
+              "_id": 4
+            }
+          },
+          "expectResult": [
+            {
+              "_id": 4,
+              "x": 12
+            }
+          ]
+        },
+        {
+          "name": "aggregate",
+          "object": "collection0",
+          "arguments": {
+            "pipeline": [
+              {
+                "$match": {
+                  "_id": 4
+                }
+              }
+            ],
+            "session": "session_mixed"
+          },
+          "expectResult": [
+            {
+              "_id": 4,
+              "x": 11
+            }
+          ]
+        },
+        {
+          "name": "distinct",
+          "object": "collection0",
+          "arguments": {
+            "fieldName": "x",
+            "filter": {
+              "_id": {
+                "$gt": 3
+              }
+            },
+            "session": "session_mixed"
+          },
+          "expectResult": [
+            11
+          ]
+        }
+      ]
+    },
+    {
+      "description": "Snapshot error is not retried",
+      "operations": [
+        {
+          "name": "find",
+          "object": "collection0",
+          "arguments": {
+            "filter": {
+              "_id": 3
+            },
+            "session": "session_invalid_snapshot"
+          },
+          "expectError": {
+            "errorCode": 72
+          }
+        }
+      ],
+      "expectEvents": [
+        {
+          "client": "client0",
+          "events": [
+            {
+              "commandStartedEvent": {
+                "command": {
+                  "find": "collection0",
+                  "filter": {
+                    "_id": 3
+                  }
+                },
+                "commandName": "find",
+                "databaseName": "database0"
+              }
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "description": "First snapshot read does send atClusterTime",
+      "operations": [
+        {
+          "name": "find",
+          "object": "collection0",
+          "arguments": {
+            "filter": {},
+            "session": "session_first_read"
+          }
+        }
+      ],
+      "expectEvents": [
+        {
+          "client": "client0",
+          "events": [
+            {
+              "commandStartedEvent": {
+                "command": {
+                  "find": "collection0",
+                  "readConcern": {
```

Comment:
Let's add commandStartedEvent assertions to all the tests, demonstrating that atClusterTime is included when appropriate.

Reply:
That's problematic as we don't have a way to obtain actual `atClusterTime`.